src/malcolm/malcolmSocket.test.js

```diff
@@ -12,15 +15,50 @@ describe('malcolm socket handler', () => {
     },
   };
 
-  const socketContainer = new MalcolmSocketContainer(socket);
+  const betterSocket = {
+    onmessage: () => {},
+    onerror: () => {},
+    onopen: () => {},
+    onclose: () => {},
+    send: payload => {
+      messages.push(payload);
+    },
+    isConnected: true,
+  };
 
   beforeEach(() => {
+    socketContainer = new MalcolmSocketContainer(socket);
+    socketContainerMK2 = new MalcolmSocketContainer(betterSocket);
     messages = [];
   });
 
+  it("sets connected if socket doesn't have property", () => {
```

Comment:
Which property doesn't it have? I'm guessing it's the socket property?
Do we need this test any more, or do we only want to test with the socket container that has a socket? I don't really understand what the 2 different scenarios are here that we're trying to cover. We probably just need to have a chat so I understand more.